lib/containers/MainContainer.js

```diff
@@ -101,42 +143,7 @@ class MainContainer extends PureComponent {
   }
 
   render() {
-    // these routes assume that the plugin metadata path
-    // is /bpanel as Links elsewhere in the codebase use
-    // /bpanel as the prefix to their paths
-    // changing this will cause things to break
-    const { match } = this.props;
-
-    // TODO: keep one source of truth for routes route side nav
-    const routes = [
-      {
-        path: `${match.url}/overview`,
-        component: props => <Overview {...props} />,
-      },
-      {
-        path: `${match.url}/send-receive`,
-        component: props => <SendReceive {...props} />,
-      },
-      {
-        path: `${match.url}/create`,
-        component: props => <CreateNew {...props} />,
-      },
-      {
-        path: `${match.url}/wallets/:type/:wallet`,
-        component: props => <WalletInfo {...props} />,
-      },
-      {
-        path: `${match.url}/multisig/:wallet/:action`,
-        component: props => <MultisigFollowUp {...props} />,
-      },
-    ];
-    const NestedViews = createNestedViews(routes);
-
-    let render;
-    if (match.isExact) render = <Welcome />;
-    else render = <NestedViews />;
```

Comment:
@tynes just FYI, this is what was causing the UI to refresh with every event. Any change, would cause NestedViews to unmount and remount even with no diff between the two app states. Also, NestedViews supports a `Home` (as well as a 404) component that can be passed in, which takes care of what you're doing with `Welcome` here.